src/components/Modals/ExploreModals/SearchModal/SearchModal.tsx

```diff
@@ -79,56 +177,113 @@ const SearchModal = ({ modalOpen, changeOpen }: ExploreModalProps) => {
+                    <GooglePlacesAutocomplete
+                      styles={{
+                        textInput: [
+                          styles.searchInput,
+                        ],
+                        listView: [styles.placesInputListView, { backgroundColor: colors.background }],
+                      }}
+                      textInputProps={{
+                        placeholderTextColor: colors.placeholder,
+                        selectionColor: colors.tint,
+                        editable: true,
+                      }}
+                      placeholder={location ? location : "Search for location"}
+                      onPress={handleGooglePlacesSearch}
+                      query={{
+                        key: GOOGLE_API_KEY,
+                        language: 'en',
+                        types: '(cities)',
+                      }} />
+                    <TouchableOpacity onPress={clearLocation}>
+                      <Text style={styles.buttonText}>Clear</Text>
+                    </TouchableOpacity>
+                  </View>
+                </Collapsable>);
+            }
+            if (item.type === 'checkin') {
+              return (
+                <Collapsable
+                  title={item.title}
+                  subtitle={checkInDate ? checkInDate : "Anyday"}
+                  contentTitle={item.title}
+                  collapsed={isCollapsed.checkin}
+                  onTouch={toggleCollapseCheckIn}
+                >
+                  <DatePicker
+                    options={{
+                      backgroundColor: colors ? BLACK : WHITE,
+                      textHeaderColor: colors ? WHITE : BLACK,
+                      textDefaultColor: colors ? WHITE : BLACK,
+                      selectedTextColor: WHITE,
+                      mainColor: TOMATO,
+                      textSecondaryColor: GREY_200,
+                      borderColor: colors ? BLACK : WHITE,
+                    }}
+                    mode="calendar"
+                    minimumDate={getCheckInMinDate()}
+                    onSelectedChange={handleCheckInChange}
+                    current={getToday()}
+                    selected={checkInDate ? checkInDate : undefined}
+                  />
+                  <TouchableOpacity onPress={clearCheckin}>
+                    <Text style={styles.buttonText}>Clear</Text>
+                  </TouchableOpacity>
+                </Collapsable>);
+            }
+            if (item.type === 'checkout') {
+              return (
+                <Collapsable
+                  title={item.title}
+                  subtitle={checkOutDate ? checkOutDate : "Anyday"}
+                  contentTitle={item.title}
+                  collapsed={isCollapsed.checkout}
+                  onTouch={toggleCollapseCheckOut}
+                >
+                  <DatePicker
+                    options={{
+                      backgroundColor: colors ? BLACK : WHITE,
```

Comment:
what do you think to create datepickerOptions memoized value like

const DatePickerStyles = usememo(() => {
 const options = {
                      selectedTextColor: WHITE,
                      mainColor: TOMATO,
                      textSecondaryColor: GREY_200,
}
const colorOption = color ? {backgroundColor: BLACK, textHeaderColor: WHITE, ...} : 
return {...option, ...colorOptions}
), [color]

moreOver, you use it twice in one file

Reply:
thanks, done